expat/lib/xmlparse.c

```diff
@@ -4979,18 +4979,18 @@ doProlog(XML_Parser parser,
     case XML_ROLE_GROUP_OPEN:
       if (parser->m_prologState.level >= parser->m_groupSize) {
         if (parser->m_groupSize) {
-          char *temp = (char *)REALLOC(parser, parser->m_groupConnector, parser->m_groupSize *= 2);
+          void *temp = REALLOC(parser, parser->m_groupConnector, parser->m_groupSize *= 2);
           if (temp == NULL) {
             parser->m_groupSize /= 2;
             return XML_ERROR_NO_MEMORY;
           }
-          parser->m_groupConnector = temp;
+          parser->m_groupConnector = (char*)temp;
           if (dtd->scaffIndex) {
-            int *temp = (int *)REALLOC(parser, dtd->scaffIndex,
+            temp = REALLOC(parser, dtd->scaffIndex,
                           parser->m_groupSize * sizeof(int));
             if (temp == NULL)
               return XML_ERROR_NO_MEMORY;
-            dtd->scaffIndex = temp;
+            dtd->scaffIndex = (int*)temp;
```

Comment:
I'm not a big fan of this change. Why share the variable if value type actually differs and they could just as well be 100% independent?

Reply:
Since we are no more using the variable temp once we assign it to m_groupConnector it will be there in stack and we can use it for allocation and assigning integer address as we defined it as void pointer. with this we can avoid extra allocation in the stack.

Reply:
Interesting point!

There is an alternative that respects types and allows use of `const` while taking the same space on the stack as re-use of `temp` does. It takes an extra pair of curly braces: A minimal demo is this:

```console
# cat main.c 
#include <stdio.h>
#include <stdlib.h>

int main() {
    {
        void * foo1 = malloc(1);
        printf("&foo1 %p\n", &foo1);
    }

    void * foo2 = malloc(1);
    printf("&foo2 %p\n", &foo2);

    return 0;
}

# gcc main.c && ./a.out
&foo1 0x7ffce82dce00
&foo2 0x7ffce82dce00
```

I have applied that approach at 0117f01906c9b52a15433314f5908c0f2aac780d over at #270  now.

Reply:
I have used above demo & below is output which I got.
foo1 and foo2 are two different variables with different address space for them.

./main
&foo1 0x7ffcb8998bc0
&foo2 0x7ffcb8998bc8

Reply:
I have tried with both GCC (8.2.0) and Clang (8.0.0) now and it seems that GCC re-uses the same address while Clang does not.

Reply:
Ideally, I would want to use two indenpent names and `const` and save `sizeof(void*)` bytes but it seems like we cannot get all at once. Any ideas? I'll need to think about it more.

Reply:
I had a chat with two other developers on Thursday about the best way to address these warnings.  There was consensus that — given the lack of a report that distinct space on the stack for these two variables, as in previous releases, would be a problem to stack size — letting stack space of 4-8 bytes make a fix less high level would be premature optimization. Furthermore, #270 does reduce stack size for GCC and with Clang being the younger of these two, there is some chance of Clang to catch up in that regard, and reduce stack usage at some point in the future for unchanged code, as well.
With that said, I will close this #269 and merge #270 to master now.